sdlib/d/gc/scanner.d

```diff
@@ -418,25 +418,26 @@ private:
 			}
 		}
 
-		if (pd.containsPointers) {
-			splitAndAddToWorklist(makeRange(e.address, e.address + e.size));
+		if (pd.containsPointers && e.usedCapacity >= PointerSize) {
+			splitAndAddToWorklist(
+				makeRange(e.address, e.address + e.usedCapacity));
 		}
 	}
 
 	void splitAndAddToWorklist(const(void*)[] range) {
 		assert(isAligned(range.ptr, PageSize),
 		       "Range is not aligned properly!");
-		assert(isAligned(range.length, PointerInPage),
-		       "Range has invalid size!");
 		assert(range.length > 0, "Cannot add empty range to the worklist!");
 
 		// Make sure we do not starve ourselves. If we do not have
 		// work in advance, then just keep some of it for ourselves.
 		if (cursor == 0) {
 			cursor = 1;
-			worklist[0] = range[0 .. PointerInPage];
+			auto myWorkSize =
+				range.length < PointerInPage ? range.length : PointerInPage;
+			worklist[0] = range[0 .. myWorkSize];
 
-			range = range[PointerInPage .. range.length];
+			range = range[myWorkSize .. range.length];
```

Comment:
You don't want to leave breadcrumbs behind, you really want to make sure whatever is left behind is at least one page in size..

Reply:
fixed.